_benchmark/user-guide/optimizing-benchmarks/performance-testing-best-practices.md

```diff
@@ -0,0 +1,277 @@
+
+```python
+# DO: Include adequate warmup
+opensearch-benchmark run --workload=geonames --warmup-time-period=300
+```
+
+The appropriate warmup time can vary depending on your specific workload and system configuration. Start with at least 5 minutes (300 seconds) and adjust as needed based on your observations.
+
+### Compare results from different environments
+
+One of the most common mistakes in performance testing is comparing results from different environments. Results obtained from a laptop or development machine are not comparable to those from a production server due to differences in hardware, network conditions, and other environmental factors.
+
+Instead, make that all comparisons are made within the same or identical environments. If you need to compare different configurations, make sure to change only one variable at a time while keeping the environment consistent.
+
+### Document environment details
+
+Proper documentation of your test environment is crucial for reproducibility and accurate analysis. Without detailed environment information, it becomes difficult to interpret results or reproduce tests in the future.
+
+Don't omit environment details from your test reports:
+
+Instead, always document comprehensive details about your test environment. This should include hardware specifications, software versions, and any relevant configuration settings. as shown in the following example:
+
+```python
+# DO: Document environment details
+def run_benchmark():
+    environment = {
+        'hardware': 'AWS m5.2xlarge',
+        'os': 'Ubuntu 20.04',
+        'kernel': '5.4.0-1018-aws',
+        'opensearch': '2.0.0',
+        'java': 'OpenJDK 11.0.11',
+        'benchmark_version': '1.0.0'
+    }
+    results = opensearch_benchmark.run()
+    return {'environment': environment, 'results': results}
+```
+
+By documenting these details, you ensure that your test results can be properly interpreted and that the tests can be reproduced if necessary.
+
+### Troubleshooting with logs
+
+When encountering issues or unexpected results, OpenSearch Benchmark logs can provide valuable insights. Here's how to effectively use logs for troubleshooting:
+
+1. Navigate to the log file: The main log file is typically located at `~/.benchmark/logs/benchmark.log`.
+
+2. Look for error messages. Search for lines containing "ERROR" or "WARNING" to identify potential issues.
+
+3. Check for performance bottlenecks. Look for entries that indicate slow operations or resource constraints.
+
+4. Review configuration details, such as logs. Logs often include information about the test configuration, which can help verify that your intended settings were applied correctly.
+
+5. Pay attention to the duration of different phases of the benchmark, including warmup and measurement periods.
+
+By carefully reviewing these logs, you can often identify the root cause of performance issues or unexpected benchmark results.
+
```

Comment:
Would also add a small sentence suggesting if users encounter unclear log errors, they should open up an issue in the OSB repository